### PR TITLE
implemented api to get method input and output metadata

### PR DESCRIPTION
**Pull request by dvoet**
This new end point parses the wdl to get input/output info. This carries type and optional status which is not present in the method config. It is also the master list; the method config is allowed to have missing entries for inputs and/or outputs.

https://broadinstitute.atlassian.net/browse/DSDEEPB-2266
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: Verify swagger UI on dev server still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiServiceSpec.scala [hunk 1]
```diff
@@ -437,6 +437,18 @@ class MethodConfigApiServiceSpec extends ApiServiceSpec {
       }
   }
 
+  it should "return 200 getting method inputs and outputs" in withTestDataApiServices { services =>
```
Comment (abaumann) [inline src/test/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiServiceSpec.scala:440]: do we need tests for the failure cases?

**Comment by abaumann** [comment]
looks good to me

File: src/test/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiServiceSpec.scala (hunk 1 above)
Comment (dvoet) [inline src/test/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiServiceSpec.scala:440]: what could possibly go wrong?

Comment (abaumann) [inline src/test/scala/org/broadinstitute/dsde/rawls/webservice/MethodConfigApiServiceSpec.scala:440]: similar to   it should "return 404 when generating a method config template from a bogus method" in withTestDataApiServices { services =>

**Comment by helgridly** [comment]
I am a bit confused but don't have much context for this. Is this essentially just a passthrough to agora? Why should this be a new endpoint

**Comment by dvoet** [comment]
no, agora does not have this end point. We get the wdl from agora but we parse it. I could see an argument for putting this in orchestration but they don't yet have a cromwell library dependency and it seems better to keep it that way for the moment. If we were to move this to orchestration I would probably push to move the methodconfigs/template end point there too.

**Pull request merged**